test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +94,11 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+
+        if(c.isMutating()){
+            assertEquals(addressBook, saveFile.load());
+        }
+
```

Comment:
Nice to see that you have updated the tests for this :) However, it would also be good to check if the command that is NOT mutating indeed doesn't save the file.